coordinator/src/main/scala/filodb.coordinator/queryengine2/QueryEngine.scala

```diff
@@ -368,13 +371,9 @@ class QueryEngine(dataset: Dataset,
     * as those are automatically prepended.
     */
   private def getColumnIDs(dataset: Dataset, cols: Seq[String]): Seq[Types.ColumnId] = {
-    val realCols = if (cols.isEmpty) Seq(dataset.options.valueColumn) else cols
```

Comment:
Why are the extra lines being removed here?  I don't see how that has to do with the change.

Reply:
Essentially, I am making the query planner not decide what column needs to be chosen. This would be done at runtime based on what range function is used etc.

I want one place where column selection happens. Without this change, it would be in multiple places.

Reply:
I moved all the column selection logic to one place, during query execution. Please see further explanation in other threads.